nexus_constructor/geometry/cylindrical_geometry.py

```diff
@@ -0,0 +1,208 @@
+        return cylinder_axis.length()
+
+    @height.setter
+    def height(self, new_height: float):
+        old_base_centre, old_base_edge, old_top_centre = self._get_cylinder_vertices()
+        cylinder_axis = old_top_centre - old_base_centre
+        old_height = cylinder_axis.length()
+        half_height_change = 0.5 * (new_height - old_height) * self.axis_direction
+        new_base_centre = qvector3d_to_numpy_array(old_base_centre - half_height_change)
+        new_base_edge = qvector3d_to_numpy_array(old_base_edge - half_height_change)
+        new_top_centre = qvector3d_to_numpy_array(old_top_centre + half_height_change)
+        vertices = np.vstack((new_base_centre, new_base_edge, new_top_centre))
+        cylinders = np.array([0, 1, 2])
+        self.file.set_field_value(self.group, "cylinders", cylinders)
+        self.file.set_field_value(self.group, "vertices", vertices)
+
+    def _get_cylinder_vertices(self) -> Tuple[QVector3D, QVector3D, QVector3D]:
+        """
+        Get the three points defining the cylinder
+        We define "base" as the end of the cylinder in the -ve axis direction
+        :return: base centre point, base edge point, top centre point
+        """
+        cylinders = self.file.get_field_value(self.group, "cylinders")
+        vertices = self.file.get_field_value(self.group, "vertices")
+        base_centre = numpy_array_to_qvector3d(vertices[cylinders[0], :])
+        base_edge = numpy_array_to_qvector3d(vertices[cylinders[1], :])
+        top_centre = numpy_array_to_qvector3d(vertices[cylinders[2], :])
+        return base_centre, base_edge, top_centre
+
+    @property
+    def radius(self) -> float:
+        base_centre, base_edge, _ = self._get_cylinder_vertices()
+        cylinder_radius = base_edge - base_centre
+        return cylinder_radius.length()
+
+    @radius.setter
+    def radius(self, new_radius: float):
+        base_centre, base_edge, _ = self._get_cylinder_vertices()
+        cylinder_radius = base_edge - base_centre
+        new_cylinder_radius = qvector3d_to_numpy_array(
+            cylinder_radius.normalized() * new_radius
+        )
+        vertices = self.file.get_field_value(self.group, "vertices")
+        vertices[1, :] = new_cylinder_radius
+        self.file.set_field_value(self.group, "vertices", vertices)
+
+    @property
+    def axis_direction(self) -> QVector3D:
+        base_centre, _, top_centre = self._get_cylinder_vertices()
+        cylinder_axis = top_centre - base_centre
+        return cylinder_axis.normalized()
+
+    @axis_direction.setter
+    def axis_direction(self, new_axis_direction: QVector3D):
+        """
+        Not required to be a unit vector, but must be non-zero
+        :param new_axis_direction: Vector describing direction of the cylinder's axis
+        """
+        validate_nonzero_qvector(new_axis_direction)
+        # TODO rotate the vertices by the angle between the new and old axis directions
```

Comment:
It seems a setter for that field it isn't needed currently, so I'll remove the method entirely. I'm too used to using python for throw-away scripts rather than application dev; need to get into the habit of TDD for this too.